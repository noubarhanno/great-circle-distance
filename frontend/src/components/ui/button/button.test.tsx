import React from "react";
import { render } from "@testing-library/react";
import Button from "./button";

describe("Button test suite", () => {
  it("should match snopshot", () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });

  it("should display spinner instead when isLoading passed", async () => {
    const { findByTestId } = render(<Button isLoading />);
    expect(await findByTestId("button-spinner")).toBeInTheDocument();
  });

  it("should render children as element", () => {
    const { getByTestId } = render(
      <Button>
        <p data-testid="button-child">button</p>
      </Button>
    );
    expect(getByTestId("button-child")).toBeInTheDocument();
  });
});
