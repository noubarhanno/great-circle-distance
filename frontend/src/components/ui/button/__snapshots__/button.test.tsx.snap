// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test suite should match snopshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1o1c5h5-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
